### ISSUE TITLE
pyxdg: AttributeError: 'MenuEntry' object has no attribute 'Type' on KDE plasma

### ISSUE THREAD

**Issue by mjlbach** [NixOS/nixpkgs#88593]
**Describe the bug**

This came up while debugging xpra on KDE. This issue has been reported [upstream](https://gitlab.freedesktop.org/xdg/pyxdg/-/issues/18), but it seems to uniquely affect KDE on nix. We communicated with the xpra author who added a mitigation for this, but it does not address the underlying issue in pyxdg. It seems the issue is due to the envvar XDG_DATA_DIRS only when it contains both `${plasma-workspace}/share` *and* `current-system/sw/share`. Those are the only two directories that seem to matter. I'm wondering if this is because some entries from plasma-workspace/share are a subset of what is symlinked into current-system/sw/share?

**To Reproduce**

1. `nix-shell -p "python3.withPackages (ps: with ps; [ pyxdg ])" --command "python -c 'from xdg.Menu import parse;parse()'"` -> Error occurs

For the following you'll need the /share directory for the installed KDE. You can get this by echo $XDG_DATA_DIRS > out.txt and searching for plasma-workspace, or on nix repl

```
❯ sudo nix repl
[sudo] password for mjlbach:
Welcome to Nix version 2.3.4. Type :? for help.

nix-repl> :l <nixpkgs>
Added 11495 variables.

nix-repl> "${plasma-workspace}"
"/nix/store/2gc5fsahhfxvffdrfz2vmr4g2askadia-plasma-workspace-5.17.5"
```

2. `nix-shell -p "python3.withPackages (ps: with ps; [ pyxdg ])" --command "XDG_DATA_DIRS=/run/current-system/sw/share:/nix/store/2gc5fsahhfxvffdrfz2vmr4g2askadia-plasma-workspace-5.17.5/share python -c 'from xdg.Menu import parse;parse()'"` -> Error occurs (change the path to point to your plasma workspace)

3.  `nix-shell -p "python3.withPackages (ps: with ps; [ pyxdg ])" --command "XDG_DATA_DIRS=/run/current-system/sw/share python -c 'from xdg.Menu import parse;parse()'"` -> Error doesn't occur

4.  `nix-shell -p "python3.withPackages (ps: with ps; [ pyxdg ])" --command "XDG_DATA_DIRS=/nix/store/2gc5fsahhfxvffdrfz2vmr4g2askadia-plasma-workspace-5.17.5/share python -c 'from xdg.Menu import parse;parse()'"` -> Error doesn't occur

pyxdg loads cleanly

**Screenshots**

**Additional context**

**Comment by mjlbach** [comment]
This should be fixed by #89570

**Comment by lourkeur** [comment]
Nixpkgs snippet to make an executable script that tests the issue
```nix
writers.writePython3 "pyxdg-test" { libraries = [ python3Packages.pyxdg ]; } ''
from xdg.Menu import parse
parse()
''
